pandas/core/arrays/datetimelike.py

```diff
@@ -313,17 +314,33 @@ def __array__(self, dtype: NpDtype | None = None) -> np.ndarray:
             return np.array(list(self), dtype=object)
         return self._ndarray
 
+    @overload
+    def __getitem__(self, item: int | np.integer) -> DTScalarOrNaT:
```

Comment:
maybe something like `ScalarIndex`

Reply:
Created a `ScalarIndexer` in next commit